### PR TITLE
refactor(components): [tabs] switch to script-setup syntax

### PR DESCRIPTION
**Pull request by freedomlang**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

1. Refactor tab-bar and tab-pane via script-setup syntax #6114

### CODE REVIEW

File: packages/components/tabs/src/tab-bar.vue [hunk 1]
```diff
@@ -1,101 +1,91 @@
...
 
-    const bar$ = ref<HTMLDivElement>()
-    const barStyle = ref()
+const ns = useNamespace('tabs')
 
-    const getBarStyle = (): CSSProperties => {
-      let offset = 0
-      let tabSize = 0
+const barRef = ref<HTMLDivElement>()
+const barStyle = ref()
```
Comment (JeremyWuuuuu) [inline packages/components/tabs/src/tab-bar.vue:31]: **Question:** can we mark this variable `barStyle` to:

```suggestion
const barStyle =  ref<CSSProperties>()
```

File: packages/components/tabs/src/tab-bar.vue [hunk 2]
```diff
@@ -1,101 +1,91 @@
...
+  async () => {
+    await nextTick()
+    update()
   },
+  { immediate: true }
+)
+useResizeObserver(barRef, () => update())
+
+defineExpose({
+  /** @description html element */
```
Comment (JeremyWuuuuu) [inline packages/components/tabs/src/tab-bar.vue:86]:
```suggestion
  /** @description tab root html element */
```

File: packages/components/tabs/src/tab-bar.vue [hunk 3]
```diff
@@ -1,101 +1,91 @@
...
+  ref: barRef,
+  /** @description update bar style */
```
Comment (JeremyWuuuuu) [inline packages/components/tabs/src/tab-bar.vue:88]:
```suggestion
  /** @description method to manually update tab bar style */
```

File: packages/components/tabs/src/tab-pane.vue [hunk 4]
```diff
@@ -25,50 +24,45 @@ import {
...
-    const active = eagerComputed(
-      () => tabsRoot.currentName.value === (props.name || index.value)
-    )
-    const paneName = computed(() => props.name || index.value)
-    const shouldBeRender = eagerComputed(
-      () => !props.lazy || loaded.value || active.value
-    )
+const instance = getCurrentInstance()!
+const tabsRoot = inject(tabsRootContextKey)
+if (!tabsRoot) throwError(COMPONENT_NAME, '<el-tabs><el-tab-pane /></el-tabs/>')
```
Comment (JeremyWuuuuu) [inline packages/components/tabs/src/tab-pane.vue:38]:
```suggestion
if (!tabsRoot) throwError(COMPONENT_NAME, 'usage: <el-tabs><el-tab-pane /></el-tabs/>')
```

**Review by jw-foss — COMMENTED** [review]
In general this LGTM, but I left a few comments for you :+1:

**Comment by sxzz** [comment]
@JeremyWuuuuu Done. PTAL.

**Review by sxzz — APPROVED** [review]

**Pull request merged**